packages/app-builder-lib/src/node-module-collector/packageManager.ts

```diff
@@ -0,0 +1,112 @@
+// copy from https://github.com/egoist/detect-package-manager/blob/main/src/index.ts
+// and merge https://github.com/egoist/detect-package-manager/pull/9 to support Monorepo
+import { promises as fs } from "fs"
+import { resolve, dirname } from "path"
+import { exec } from "child_process"
+import { promisify } from "util"
+
+const execa = promisify(exec)
+
+export type PM = "npm" | "yarn" | "pnpm" | "bun"
+
+/**
+ * Check if a path exists
+ */
+async function pathExists(p: string) {
+  try {
+    await fs.access(p)
+    return true
+  } catch {
+    return false
+  }
+}
+
+const cache = new Map()
```

Comment:
Might be good to rename this from generic `cache` as to what its purpose is serving, especially since it's a file-level var. e.g. something like `const packageManagerVersions = new Map<string, string>()`